index.c

```diff
@@ -3091,15 +3090,17 @@ int mutt_index_menu(void)
       case OP_VIEW_RAW_MESSAGE:
       {
         /* TODO split this into 3 cases? */
-        CHECK_MSGCOUNT;
-        CHECK_VISIBLE;
-        CHECK_ATTACH;
+        if (!prereq(Context, menu, CHECK_IN_MAILBOX | CHECK_MSGCOUNT | CHECK_VISIBLE | CHECK_ATTACH))
+          break;
         bool edit;
         if (op == OP_EDIT_RAW_MESSAGE)
         {
-          CHECK_READONLY;
-          /* L10N: CHECK_ACL */
-          CHECK_ACL(MUTT_ACL_INSERT, _("Cannot edit message"));
+          if (!prereq(Context, menu, CHECK_READONLY) ||
+              /* L10N: CHECK_ACL */
```

Comment:
I think we should move the comment above the conditional, it makes it easier to see the entire conditional.